modules/gapi/samples/face_detection.cpp

```diff
@@ -0,0 +1,764 @@
+                    (faces[tmpIdx].bbox.y2 - faces[tmpIdx].bbox.y1 + 1);
+                double overlap = 0.0;
+                if (useMin) {
+                    overlap = interArea / std::min(area1, area2);
+                }
+                else {
+                    overlap = interArea / (area1 + area2 - interArea);
+                }
+                if (overlap <= threshold) {
+                    indices.push_back(tmpIdx);
+                }
+            }
+        }
+        return facesNMS;
+    }
+};
+
+const double P_NET_WINDOW_SIZE = 12.0;
+const double P_NET_STRIDE = 2.0;
+
+std::vector<Face> buildFaces(const cv::Mat& scores,
+    const cv::Mat& regressions,
+    const double scaleFactor,
+    const double threshold) {
+
+    auto w = scores.size[3];
+    auto h = scores.size[2];
+    auto size = w * h;
+
+    const float* scores_data = scores.ptr<float>();
+    scores_data += size;
+
+    const float* reg_data = regressions.ptr<float>();
+
+    std::vector<Face> boxes;
+
+    for (int i = 0; i < size; i++) {
+        if (scores_data[i] >= (threshold)) {
+            int y = i / w;
+            int x = i - w * y;
+
+            Face faceInfo;
+            BBox& faceBox = faceInfo.bbox;
+
+            faceBox.x1 = (static_cast<double>(x) * P_NET_STRIDE) / scaleFactor;
+            faceBox.y1 = (static_cast<double>(y) * P_NET_STRIDE) / scaleFactor;
+            faceBox.x2 = (static_cast<double>(x) * P_NET_STRIDE + P_NET_WINDOW_SIZE - 1.f) / scaleFactor;
+            faceBox.y2 = (static_cast<double>(y) * P_NET_STRIDE + P_NET_WINDOW_SIZE - 1.f) / scaleFactor;
+            faceInfo.regression[0] = reg_data[i];
+            faceInfo.regression[1] = reg_data[i + size];
+            faceInfo.regression[2] = reg_data[i + 2 * size];
+            faceInfo.regression[3] = reg_data[i + 3 * size];
+            faceInfo.score = scores_data[i];
+            boxes.push_back(faceInfo);
+        }
+    }
+
+    return boxes;
+}
+
+// Define networks for this sample
+using GMat2 = std::tuple<cv::GMat, cv::GMat>;
+using GMat3 = std::tuple<cv::GMat, cv::GMat, cv::GMat>;
+using GMats = cv::GArray<cv::GMat>;
+using GRects = cv::GArray<cv::Rect>;
+using GSize = cv::GOpaque<cv::Size>;
+
+
+G_API_NET(MTCNNRefinement,
+          <GMat2(cv::GMat)>,
+          "sample.custom.mtcnn_refinement");
+
+G_API_NET(MTCNNOutput,
+          <GMat3(cv::GMat)>,
+          "sample.custom.mtcnn_output");
+
+using GFaces = cv::GArray<Face>;
+G_API_OP(BuildFaces,
+         <GFaces(cv::GMat, cv::GMat, double, double)>,
+         "sample.custom.mtcnn.build_faces") {
+            static cv::GArrayDesc outMeta(const cv::GMatDesc&,
+            const cv::GMatDesc&,
+            double,
+            double) {
+            return cv::empty_array_desc();
+    }
+};
+
+G_API_OP(RunNMS,
+         <GFaces(GFaces, double, bool)>,
+         "sample.custom.mtcnn.run_nms") {
+         static cv::GArrayDesc outMeta(const cv::GArrayDesc&,
+         double, bool) {
```

Comment:
Needs to move under `outMeta(`
Code becomes more readable.

Reply:
Done